improver_tests/ensemble_copula_coupling/test_EnsembleReordering.py

```diff
@@ -414,6 +512,56 @@ def test_basic(self):
         )
         self.assertArrayAlmostEqual(result.data, expected_data)
 
+    @ManageWarnings(ignored_messages=["Only a single cube so no differences"])
+    def test_basic_masked_input_data(self):
+        """
+        Test that the plugin returns an iris.cube.Cube, the cube has a
+        realization coordinate with specific realization numbers and is
+        correctly re-ordered to match the source realizations, when the
+        input data is masked.
+        """
+        # Assuming inptu data and raw ensemble are masked in the same way.
+        self.raw_cube.data[:, 0, 0] = np.nan
+        self.raw_cube.data = np.ma.masked_invalid(self.raw_cube.data)
+        self.post_processed_percentiles.data[:, 0, 0] = np.nan
+        self.post_processed_percentiles.data = np.ma.masked_invalid(
+            self.post_processed_percentiles.data
+        )
+        expected_data = self.raw_cube.data.copy()
+        result = Plugin().process(self.post_processed_percentiles, self.raw_cube)
+        self.assertIsInstance(result, Cube)
+        self.assertTrue(result.coords("realization"))
+        self.assertEqual(
+            result.coord("realization"), self.raw_cube.coord("realization")
+        )
+        self.assertArrayAlmostEqual(result.data, expected_data)
+        self.assertArrayAlmostEqual(result.data.mask, expected_data.mask)
+
+    @ManageWarnings(ignored_messages=["Only a single cube so no differences"])
+    def test_basic_masked_input_data_not_nans(self):
+        """
+        Test that the plugin returns an iris.cube.Cube, the cube has a
+        realization coordinate with specific realization numbers and is
+        correctly re-ordered to match the source realizations, when the
+        input data is masked.
```

Comment:
This doc-string is identical to the previous test. It should clarify that in this case the data contains no nans.